### ISSUE TITLE
error: bundling failed: Error: Unable to resolve module `./navigators/createContainedStackNavigator`

### ISSUE THREAD

**Issue by dwisetiyadi** [react-navigation/react-navigation#4895]
## Unable to resolve module `./navigators/createContainedStackNavigator`

Can not run when run
`react-native ios`

---

### Current Behavior

when run
`react-native run-ios`

i have this error message:

error: bundling failed: Error: Unable to resolve module `./navigators/createContainedStackNavigator` from `~/node_modules/react-navigation/src/react-navigation.js`: The module `./navigators/createContainedStackNavigator` could not be found from `~/node_modules/react-navigation/src/react-navigation.js`. Indeed, none of these files exist:

### Expected Behavior

Should run my app on iOS Simulator

### How to reproduce

- npm i --save react-navigation
- react-native run-ios

### My Environment

| software         | version
| ---------------- | -------
| react-navigation | 2.12.1
| react-native     | 0.56.0
| node             | v10.9.0
| npm or yarn      | 6.2.0

**Comment by welladam** [comment]
I'm get the same error, happened suddenly without me having updated anything!

software | version
-- | --
react-navigation | 1.5.13
react-native | 0.52.0
node | v10.0.0
npm or yarn | 6.0.1

**Comment by brentvatne** [comment]
hello i suggest deleting node_modules, deleting your lock files, and running npm install again

**Comment by dwisetiyadi** [comment]
I did this, over and over, but the problem did'nt resolve

**Comment by raffij** [comment]
@dwisetiyadi it's to do with the new version refactoring the StackNavigator require.

Removing node_modules, deleting app and closing the build console didn't help.

I needed to reset the react-native cache `npm start -- --reset-cache`

**Comment by GunnarAK** [comment]
Performing `npm start -- --reset-cache` on its own did not do the trick for me.

But performing the following did succeed:

1. Delete node_modules & package-lock.json
2. Delete app from phone
3. run `npm start -- --reset-cache`
4. run app

**Comment by laurentsmohr** [comment]
@GunnarAK how does your npm start script look like?

**Comment by beckinfonet** [comment]
> Performing `npm start -- --reset-cache` on its own did not do the trick for me.
> 
> But performing the following did succeed:
> 
> 1. Delete node_modules & package-lock.json
> 2. Delete app from phone
> 3. run `npm start -- --reset-cache`
> 4. run app

Thanks man, fixed it!

**Comment by kunalshah4345** [comment]
change version of react-navigation  to 1.6.1 in package.json file
Delete node_modules & package-lock.json
Delete app from phone
run npm start -- --reset-cache
run app

**Comment by JGeraldoLima** [comment]
I can confirm that only removing **node_modules** path does solve the problem. 

If you're getting an error with pattern `error: bundling failed: Error: Unable to resolve module '<lib>' from '<lib file>': Module '<module>' does not exist in the Haste module map`, check you **react** and **react-native** versions or the **last added lib** on **package.json**, maybe some of them are crashed.

BUT, if someone of your team are not getting this error, maybe it is a cache problem. Just follow @GunnarAK tips and save your life, like I did. Thank you so much @GunnarAK!

**Comment by GunnarAK** [comment]
@laurentsmohr I will have a look at this shortly

**Comment by GunnarAK** [comment]
@laurentsmohr Here it is:

`"start": "react-native start",`

**Comment by jdevey** [comment]
I scoured the internet for at least six hours trying to find a solution to this problem, and none of them worked. Not even a little bit.

At one point I was so desperate I started renaming some directories, and out of dumb luck I found that changing a directory with a capitalized name to a lower-case name fixed the issue. I'm not entirely sure why this fixed it, but I suspect it may have to do with npm.

TL;DR change the imports that are giving you issues so the directory names are lower-case.

**Comment by GunnarAK** [comment]
From the docs (found at https://github.com/nwjs/nw.js/wiki/Manifest-format#name)
```name
(string) the name of the package. This must be a unique, lowercase alpha-numeric name without spaces. It may include "." or "_" or "-" characters. It is otherwise opaque.

name should be globally unique since node-webkit will store app's data under the directory named name.```

**Comment by shayanaithal** [comment]
![screenshot from 2019-02-07 12-00-58](https://user-images.githubusercontent.com/47058636/52394406-2e4eef80-2ad0-11e9-8da4-1ba06cb3b322.png)
i tryd everything still getting this error

**Comment by us-22** [comment]
+1 

I'm facing the same issue? any luck resolving it?

**Comment by jameskip** [comment]
I am using Yarn and was unable to reproduce the above solutions. I was able to resolve this error by running:

`yarn cache clean`
